swift/generate-short-url/index.swift

```diff
@@ -0,0 +1,98 @@
+import AsyncHTTPClient
+import Foundation
+
+func main(req: RequestValue, res: RequestResponse) async throws -> RequestResponse {
+    var status: Bool = false
+    var isBitly: Bool = false
+    var isTinyUrl: Bool = false
+
+    var type: String = ""
+    var url: String = ""
+    var tokenKey: String = ""
+    var token: String = ""
+    var result: [String : Any?]
+
+    var providerType: ProviderType = .Invalid
+    let network = Network()
+
+    result = [
+        "status": status,
+        "message": "Missing payload",
+    ]
+
+    if !req.payload.isEmpty,
+        let data = req.payload.data(using: .utf8),
+        let payload = try? JSONSerialization.jsonObject(with: data, options: []) as? [String: Any] {
+            type = payload["provider"] as? String ?? ""
+            url = payload["url"] as? String ?? ""
+    } else {
+        return res.json(data: result)
+    }
+
+    if type.isEmpty {
+        result = [
+            "status": status,
+            "message": "Empty provider type",
+        ]
+        return res.json(data: result)
+    }
+
+    if url.isEmpty {
+        result = [
+            "status": status,
+            "message": "Empty url to be shorten",
+        ]
+        return res.json(data: result)
+    }
+
+    isBitly = type.lowercased().elementsEqual("bitly")
+    isTinyUrl = type.lowercased().elementsEqual("tinyurl")
+
+    providerType = isBitly ? .Bitly : isTinyUrl ? .TinyUrl : .Invalid
+    tokenKey = isBitly ? TokenType.Bitly.rawValue : isTinyUrl ? TokenType.TinyUrl.rawValue : ""
+    token = req.env[tokenKey] ?? ""
```

Comment:
Make sure to replace `req.env` with `req.variables`:

```suggestion
    token = req.variables[tokenKey] ?? ""
```